internal/internal_task_pollers.go

```diff
@@ -346,7 +350,13 @@ func (wtp *workflowTaskPoller) processWorkflowTask(task *workflowTask) error {
 		return err
 	}
 	var taskErr error
-	defer func() { wfctx.Unlock(taskErr) }()
+	defer func() {
+		// This check exists to avoid a panic in unit tests where this is nil. In non-mocked
```

Comment:
Can we just make sure it is not nil in unit tests as well or is it really hard? I don't like adding checks like this in production code just to make tests a bit simpler

Reply:
Well, that worked out. Turns out it no longer happens anymore anyway, which must've been a result of refactoring along the way.